docs/admin/compatibility_matrix.md

```diff
@@ -0,0 +1,10 @@
+# Compatibility Matrix
+
+!!! warning "Developer Note - Remove Me!"
+    Explain how the release models of the plugin and of Nautobot work together, how releases are supported, how features and older releases are deprecated etc.
```

Comment:
What's the plan for addressing all these "Remove Me!" items and generally making sure all these newly added docs are actually relevant/accurate to the app? Part of this PR or future work?

Reply:
Will fix these

Reply:
@cmsirbu What do you think about the documentation? Does it make sense to rewrite it from README and if so, to do it in the separater PR?

Reply:
After the settlement with @cmsirbu , removed all changes to docs. Kept updated docs requirements in `pyproject.toml` and `docs/requirements`. Docs update should be done here: https://github.com/nautobot/nautobot-plugin-secrets-providers/pull/54